### PR TITLE
automation: add save as button

### PR DESCRIPTION
**Pull request by Siddharth3330**
Added
- A save-as button next to save button in automation toolbar in order to save the plan.

Code change
- Made a new method named savePlan() and copied lambda value form getSavePlanButton() method in it to be reused in getSaveAsPlanButton() and getSavePlanButton() methods with adequate flag values.
- Updated Messages.properties and CHANGELOG.md files regarding save-as button. 

Fix zaproxy/zaproxy#7178.

### CODE REVIEW

**Comment by kingthorin** [comment]
To address the DCO requirement you'll need to sign-off the commit(s):

- https://github.com/zaproxy/zaproxy/blob/main/CONTRIBUTING.md#developer-certificate-of-origin
- https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---signoff

**Comment by kingthorin** [comment]
Have a look here: https://github.com/zaproxy/zap-extensions/tree/main/addOns/automation/src/main/resources/org/zaproxy/addon/automation/resources for other icons and search their names in *.java to see how they’re used.

File: addOns/automation/src/main/java/org/zaproxy/addon/automation/gui/AutomationPanel.java [hunk 1]
```diff
@@ -203,6 +208,63 @@ private JButton getRunPlanButton() {
         return runPlanButton;
     }
 
+    private void savePlan(boolean promptForFile) {
+        if (currentPlan == null) {
+            return;
+        }
+        try {
+            if (promptForFile) {      //promptForFile replaced currentPlan.getFile() == null
```
Comment (thc202) [inline addOns/automation/src/main/java/org/zaproxy/addon/automation/gui/AutomationPanel.java:216]: The comment should be removed.

File: addOns/automation/src/main/java/org/zaproxy/addon/automation/gui/AutomationPanel.java [hunk 2]
```diff
@@ -730,6 +751,13 @@ private void updateSaveButton(Event event, boolean enable) {
         }
     }
 
+    private void updateSaveAsButton(Event event, boolean enable) {
```
Comment (thc202) [inline addOns/automation/src/main/java/org/zaproxy/addon/automation/gui/AutomationPanel.java:754]: This method should not be needed actually, we just want to change the enabled state when the plan changes which is already covered by `setCurrentPlan`.

File: addOns/automation/src/main/java/org/zaproxy/addon/automation/gui/AutomationPanel.java [hunk 3]
```diff
@@ -558,6 +578,7 @@ public void setCurrentPlan(AutomationPlan plan) {
         getRunPlanButton().setEnabled(currentPlan != null);
         getAddJobButton().setEnabled(currentPlan != null);
         getSavePlanButton().setEnabled(false);
+        getSaveAsPlanButton().setEnabled(false);
```
Comment (thc202) [inline addOns/automation/src/main/java/org/zaproxy/addon/automation/gui/AutomationPanel.java:581]: `currentPlan != null`, we want it to be enabled when there's a plan.

File: addOns/automation/CHANGELOG.md [hunk 4]
```diff
@@ -4,7 +4,8 @@ All notable changes to this add-on will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 
 ## Unreleased
-
+### Added
+- Save-as button next to the save button for saving plans.
```
Comment (thc202) [inline addOns/automation/CHANGELOG.md:8]: e.g. `Add Save As button that allows to save the automation plan to a different file (Issue 7178).`

**Comment by thc202** [comment]
Seems to need a `spotlessApply`.

File: addOns/automation/src/main/resources/org/zaproxy/addon/automation/resources/Messages.properties [hunk 5]
```diff
@@ -156,6 +156,7 @@ automation.dialog.plan.load = Load Plan...
 automation.dialog.plan.new = New Plan...
 automation.dialog.plan.run = Run Plan...
 automation.dialog.plan.save = Save Plan...
+automation.dialog.plan.save-as = Save-as Plan...
```
Comment (thc202) [inline addOns/automation/src/main/resources/org/zaproxy/addon/automation/resources/Messages.properties:159]: `Save Plan As...`

**Comment by thc202** [comment]
Could you squash the commits?

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Review by kingthorin — APPROVED** [review]

**Comment by Siddharth3330** [comment]
Siddharth Singh, https://twitter.com/runny_sunny, https://github.com/Siddharth3330

**Pull request merged**